components/director/pkg/correlation/middleware.go

```diff
@@ -1,43 +1,84 @@
+/*
+ * Copyright 2020 The Compass Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package correlation
 
 import (
 	"context"
 	"net/http"
+
+	"github.com/google/uuid"
 )
 
 const (
-	ContextField       = "CorrelationID"
+	HeadersContextKey  = "CorrelationHeaders"
 	RequestIDHeaderKey = "x-request-id"
 )
 
-var headerKeys = []string{
-	RequestIDHeaderKey,
-	"X-Request-ID",
-	"X-Request-Id",
-	"X-Correlation-ID",
-	"X-CorrelationID",
-	"X-ForRequest-ID"}
+// headerKeys are the expected headers that are used for distributed tracing.
+var headerKeys = []string{"x-request-id", "x-b3-traceid", "x-b3-spanid", "x-b3-parentspanid", "x-b3-sampled", "x-b3-flags", "b3"}
```

Comment:
Should we make these configurable as well?

Reply:
maybe we could. just one question - how did the tracing work properly before as we didnt propagate these ?

Reply:
It worked only for `gateway` -> `director`/`connector` requests because the gateway only proxies the request with all of its headers.

The values in the lua script are the correlation IDs that might come from outside components,
The values here are mentioned in the [Istio documentation](https://istio.io/latest/faq/distributed-tracing/#how-to-support-tracing) and are used for tracing within the cluster.
I'll add a new config entry then.

Reply:
i guess this explains why there was no arrow in kiali between director and connector. Basically with this fix we should get this arrow, right ?
![image](https://user-images.githubusercontent.com/8989462/97879174-5b254a80-1d28-11eb-880b-8964778a7d1c.png)

Reply:
Exactly!

Reply:
![image](https://user-images.githubusercontent.com/9128192/97957379-b51f2200-1db3-11eb-90ba-4dfb0586241a.png)